dll/ntdll/ldr/ldrutils.c

```diff
@@ -1871,6 +1871,7 @@ LdrpSearchPath(IN PWCHAR *SearchPath,
                            Buffer);
             }
 
+            ASSERT(BufEnd != NULL);
```

Comment:
I think what we're really missing here is a `BufEnd = Buffer` assignment before the start of the loop.

Reply:
That makes sense